tests/backends/test_Windows.py

```diff
@@ -58,9 +29,25 @@ def test_read_utf8_password(self):
         username = "keyring"
         password = "utf8-test" + UNICODE_CHARS
 
-        self.set_utf8_password(service, username, password)
+        self.keyring.set_password(service, username, password)
         assert self.keyring.get_password(service, username) == password
 
+    def test_long_password_nice_error(self):
+        self.keyring.set_password('system', 'user', 'x' * 10000)
+        self.keyring.delete_password('system', 'user')
+
+    def test_long_password_too_long_nice_error(self):
+        try:
+            self.keyring.set_password('system', 'user', 'x' * (2**20 + 1))
+            self.keyring.delete_password('system', 'user')
+        except ValueError as e:
+            if e.args[0] == 2**20:
+                assert True
+            else:
+                assert False
```

Comment:
Eww. Just use `assert e.args[0] == 2**20`.

Reply:
I'm embarrassed.  Of course your way is a lot nicer!